tests/test_core.py

```diff
@@ -1061,6 +1061,134 @@ def test_select_P_ABBA_val_inf():
     npt.assert_almost_equal(ref, comp)
 
 
+def test_merge_topk_PI():
+    # `assume_unique = True`
```

Comment:
replace the comment with:
"# test case: there are no overlapping values between each row of IA, e.g. `IA[i]` and its corresponding row in IB, i.e. `IB[i]`"

Reply:
Then I think you should split this unit test up into smaller tests. One called `test_merge_topk_PI_without_overlap` and `test_merge_topk_PI_with_overlap` and then add this comment.

Reply:
Technically, each test should clearly test ONE thing and test it clearly.

Reply:
I see. So, I will create two test functions, each dedicated to one thing. Thanks for the input :)